atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/keyvalue/api/cache/LockWatchValueScopingCacheImpl.java

```diff
@@ -213,11 +214,9 @@ private synchronized boolean isNewEvent(LockWatchEvent event) {
                 .orElse(true);
     }
 
-    private synchronized void assertNoSnapshotsMissing(Set<Sequence> sequences) {
-        if (sequences.stream()
-                .map(snapshotStore::getSnapshotForSequence)
-                .anyMatch(maybeSnapshot -> !maybeSnapshot.isPresent())) {
-            throw new TransactionLockWatchFailedException("snapshots were not taken for all sequences; this update "
+    private synchronized void assertNoSnapshotsMissing(Collection<StartTimestamp> timestamps) {
+        if (timestamps.stream().map(snapshotStore::getSnapshot).anyMatch(Optional::isEmpty)) {
+            throw new TransactionLockWatchFailedException("snapshots were not taken for all timestamps; this update "
                     + "must have been lost and is now too old to process. Transactions should be retried.");
```

Comment:
this was a bit of a weird check, as we actually didn't mandate that all sequences had snapshots, but rather only sequences associated with timestamps